### PR TITLE
🧪 [CW-22356] test: generate test report

### PR DESCRIPTION
**Pull request by jin860315**
1. 本次新增 `run-test-and-generate-report.yml` 檔與 `auto_test.yml` 執行測試的方式一樣
2. 以及最後的產出報告，有先透過[測試 repo](https://github.com/CoolBitX-Technology/brian-test/actions/runs/11142327106/job/30965045081) 驗證可以產出測試報告

----

*Checklist:*

- README.md includes:
  - [ ] The details of signing data and transaction data, this includes parameters.
  - [ ] Official docs or white paper.
  - [ ] Website or api can query assets and broadcast transaction.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

 
 **PR Summary by Typo**
------------

 **Summary**

This pull request introduces two new GitHub Actions workflows: `run-test-and-generate-report.yml` and handles JUnit test reports publication. The first workflow automates testing and report generation for modified package.json files, while the second one publishes JUnit reports as artifacts and on GitHub Pages.

**Key Points**

1. Adds `run-test-and-generate-report.yml` for testing and report generation.
2. Installs dependencies, runs tests, and publishes JUnit reports.
3. Introduces "scripts/handling-report.sh" for JUnit report processing. 

 <h6>To turn off PR summary, please visit <a href="https://app.typoapp.io/settings/dev-analytics/notification?tab=codeHealth">Notification settings</a>.</h6>

### CODE REVIEW

File: .github/workflows/run-test-and-generate-report.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
+name: Run Test and Generate Report
```
Comment (tp26610) [inline .github/workflows/run-test-and-generate-report.yml:1]: Run 想留給手動執行的 action 當作特殊前綴，改 `Execute Test and Generate Report` 如何？

File: scripts/handing-report.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# 移動測試報告
+for dir in packages/coin-* packages/core; do
+  if [ -d "$dir" ] && [ -f "$dir/junit.xml" ]; then
+    coin_name=$(basename "$dir")
+    sed 's/&quot;/"/g' "$dir/junit.xml" >"scripts/reports/junit/${coin_name}-junit.xml" || echo "Failed to process $dir/junit.xml"
+  fi
+done
+
+# 移除所有 junit.xml 文件
+find packages -type f -name 'junit.xml' -exec rm -f {} +
```
Comment (tp26610) [inline scripts/handing-report.sh:13]: 加個跳行結尾

**Comment by tp26610** [comment]
我看網頁 https://coolbitx-technology.github.io/brian-test/ 顯示 404。是因為我連錯網址嗎？

File: .github/workflows/run-test-and-generate-report.yml (hunk 1 above)
Comment (jin860315) [reply .github/workflows/run-test-and-generate-report.yml]: 改改～
f9b1320efc8f923349806d8891b8ae35a585dc73

File: scripts/handing-report.sh (hunk 2 above)
Comment (jin860315) [reply scripts/handing-report.sh]: 改改～
f9b1320efc8f923349806d8891b8ae35a585dc73

Comment (jin860315) [reply scripts/handing-report.sh]: 報告的話，可以幫我看這個 url ～
https://coolbitx-technology.github.io/brian-test/all-services/23/

**Comment by cbx-tech** [comment]
Task linked: [CW-22356 SDK 跑完測試後可呈現測試報告](https://app.clickup.com/t/25577573/CW-22356)

**Review by tp26610 — APPROVED** [review]

**Review by tp26610 — APPROVED** [review]

**Review by a0979470582 — APPROVED** [review]

**Pull request merged**